Sources/Host/Darwin/X86_64/MachX86_64.cpp

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// Copyright (c) 2015, Corentin Derbois <cderbois@gmail.com>
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#include "DebugServer2/Host/Darwin/Mach.h"
+#include "DebugServer2/Host/Platform.h"
+#include "DebugServer2/Utils/Log.h"
+
+#include <cstring>
+#include <mach/mach.h>
+#include <mach/mach_vm.h>
+#include <mach/thread_info.h>
+#include <sys/ptrace.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/user.h>
+
+#define super ds2::Host::POSIX::Mach
+
+namespace ds2 {
+namespace Host {
+namespace Darwin {
+
+ErrorCode Mach::readCPUState(ProcessThreadId const &ptid,
+                             ProcessInfo const &pinfo,
+                             Architecture::CPUState &ds2_state) {
+  pid_t tid;
+
+  if (!ptid.valid())
+    return kErrorInvalidArgument;
+
+  if (!(ptid.tid <= kAnyThreadId)) {
+    tid = ptid.tid;
+  } else {
+    tid = ptid.pid;
+  }
+
+  thread_t thread = getMachThread(ptid.pid, tid);
+  if (thread == THREAD_NULL)
+    return kErrorProcessNotFound;
+
+  kern_return_t kret;
+  mach_msg_type_number_t stateCount = x86_THREAD_STATE_COUNT;
+  x86_thread_state_t state;
+
+  kret = thread_get_state(thread, x86_THREAD_STATE, (thread_state_t)&state,
+                          &stateCount);
+  if (kret != KERN_SUCCESS) {
+    DS2LOG(Error, "Fail to get the reg of the pid: %s",
+           mach_error_string(kret));
+    return kErrorInvalidArgument;
+  }
+
+  ds2_state.is32 = false;
+  ds2_state.state64.gp.rax = state.uts.ts64.__rax;
```

Comment:
For this big block of registers, look at `user_to_state64` and `state64_to_user`. If the macros are too much of a pain to configure/use, I'm fine with checking in the current version of the register copy; I'll make the change myself later on.